test/lsp/lsp_completion.ml

```diff
@@ -30,26 +34,154 @@ let completion_positions (doc, positions) : string -> unit =
       location_as_srcloc#pp location
       Fmt.(option ~none:nop (string ++ sp)) key
       position.line position.character;
-    match LSP.Request.completion server params with
+    let noeagerlist = match LSP.Request.completion ~eager:false server params with
+      | None ->
+        Pretty.out "Failed completion@."; []
+      | Some `CompletionList { items; _ } when items == [] ->
+        Pretty.out "Empty completion list@."; []
+      | Some `CompletionList { items; _ } ->
+        let items = map_to_string items in
+        Pretty.out "List of completions (%d entries): [%a]\n"
+          (List.length items)
+          (Fmt.list ~sep:(Fmt.any ";") Fmt.string)
+          items;
+        items
+    in
+    match LSP.Request.completion ~eager:true server params with
     | None ->
-        Pretty.out "Failed completion@."
+      Pretty.out "Failed eager-completion@."
+    | Some `CompletionList { items; _ }
+      when List.equal String.equal (map_to_string items) noeagerlist ->
+      ()
     | Some `CompletionList { items; _ } when items == [] ->
-        Pretty.out "Empty completion list@."
+      Pretty.out "Empty eager-completion list@."
     | Some `CompletionList { items; _ } ->
-        let pp_comp_item ppf (item: CompletionItem.t) =
-          Fmt.string ppf item.label in
-        Pretty.out "List of completions (%d entries): [%a]\n"
-          (List.length items) (Fmt.list ~sep:(Fmt.any ";") pp_comp_item) items;
-
+      Pretty.out "List of eager-completion (%d entries): [%a]\n"
```

Comment:
`"@[<hov 2>Eager (%d entries):@;[%a]@\n"`

and then `sep:Fmt.semicolon` below.